### PR TITLE
Support COM events

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Add support for COM events
Add basic tests for COM events

See https://github.com/dotnet/coreclr/issues/22250

cc @jeffschwMSFT @jkotas @onovotny

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComEventsMethod.cs [hunk 1]
```diff
@@ -2,96 +2,139 @@
...
 using System.Reflection;
 
-
 namespace System.Runtime.InteropServices
 {
-    // see code:ComEventsHelper#ComEventsArchitecture
+    /// <summary>
+    /// Part of ComEventHelpers APIs which allow binding
+    /// managed delegates to COM's connection point based events.
+    /// see code:ComEventsHelper#ComEventsArchitecture
```
Comment (jkoritzinsky) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComEventsMethod.cs]: There's a few of these old code references to what I assume are framework docs. We should either remove these docs or add the doc to somewhere that the community can read them like docs or the BOTR.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Variant.cs [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+        private struct Record
+        {
+            public IntPtr _record;
+            public IntPtr _recordInfo;
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        private struct UnionTypes
+        {
+            [FieldOffset(0)] public SByte _i1;
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Variant.cs]: Nope. Will use keywords instead.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComEventsMethod.cs (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComEventsMethod.cs]: Those still exist. See `ComEventsHelper.cs`

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Ubuntu x64 Formatting please

**Pull request merged**